cdci_data_analysis/analysis/instrument.py

```diff
@@ -138,19 +138,17 @@ def set_pars_from_dic(self, par_dic, verbose=False):
         updated_par_dic = par_dic.copy()
         
         for par in param_list:
-            # this is required because in some cases a parameter is set without a name (eg UserCatalog),
-            # or they don't have to set (eg scw_list)
-
             self.logger.info("before normalizing, set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s",
                              par, par.name, par.value, par_dic.get(par.name, None))
 
+            # this is required because in some cases a parameter is set without a name (eg UserCatalog),
+            # or they don't have to set (eg scw_list)
             if par.name is not None and par.name not in params_not_to_be_included:
-                # set the value for par in to a default format,
+                # set the value for par to a default format,
                 # or to a default value if this is not included within the request
-                
                 updated_par_dic[par.name] = par.set_value_from_form(par_dic, verbose=verbose)
 
-                if par.default_units is not None:
+                if par.units_name is not None and par.default_units is not None:
```

Comment:
what if only one of these two conditions is satisfied? This would be a problem in constructing, e.g. in plugin. Will there be an exception when constructing?

Reply:
are you referring to a case of a `Parameter` that has no `default_units`? but it has a `units_name`?

well, yes, in that case yes probably an exception is necessary

Reply:
yes. And also dor the opposite (`default_units` but no `units_name`) - maybe a warning.

Reply:
is it really a problem if `units_name` is not specified for a `Parameter` ?

Reply:
it's  not a big problem, but it's probably unintended. It's good to add it. So a warning - not an exception.